api/rql/prune.go

```diff
@@ -0,0 +1,135 @@
+package rql
```

Comment:
See https://github.com/puppetlabs/wash/tree/master/cmd/internal/find/types, specifically the `entrySchema` files as much of this was copied from there. Only addition was modifying the schema paths to match what Kind primary expects.